src/Core/src/Handlers/Picker/PickerHandler.Android.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Specialized;
+using System.Linq;
+using Android.App;
+using Android.Text;
+using Android.Text.Style;
+using AResource = Android.Resource;
+
+namespace Microsoft.Maui.Handlers
+{
+	public partial class PickerHandler : AbstractViewHandler<IPicker, NativePicker>
+	{
+		AlertDialog? _dialog;
+
+		protected override NativePicker CreateNativeView() =>
+			new NativePicker(Context);
+
+		protected override void ConnectHandler(NativePicker nativeView)
+		{
+			nativeView.FocusChange += OnFocusChange;
+			nativeView.Click += OnClick;
+
+			if (VirtualView != null && VirtualView.Items is INotifyCollectionChanged notifyCollection)
+				notifyCollection.CollectionChanged += OnCollectionChanged;
+
+			base.ConnectHandler(nativeView);
+		}
+
+		protected override void DisconnectHandler(NativePicker nativeView)
+		{
+			nativeView.FocusChange -= OnFocusChange;
+			nativeView.Click -= OnClick;
+
+			if (VirtualView != null && VirtualView.Items is INotifyCollectionChanged notifyCollection)
+				notifyCollection.CollectionChanged -= OnCollectionChanged;
+
+			base.DisconnectHandler(nativeView);
+		}
+		public static void MapTitle(PickerHandler handler, IPicker picker)
+		{
+			handler.TypedNativeView?.UpdateTitle(picker);
+		}
+
+		public static void MapSelectedIndex(PickerHandler handler, IPicker picker)
+		{
+			handler.TypedNativeView?.UpdateSelectedIndex(picker);
+		}
+
+		void OnFocusChange(object? sender, global::Android.Views.View.FocusChangeEventArgs e)
+		{
+			if (TypedNativeView == null)
+				return;
+
+			if (e.HasFocus)
+			{
+				if (TypedNativeView.Clickable)
+					TypedNativeView.CallOnClick();
+				else
+					OnClick(TypedNativeView, EventArgs.Empty);
+			}
+			else if (_dialog != null)
+			{
+				_dialog.Hide();
+				TypedNativeView.ClearFocus();
+				_dialog = null;
+			}
+		}
+
+		void OnClick(object? sender, EventArgs e)
+		{
+			if (VirtualView != null && _dialog == null)
```

Comment:
nit: maybe invert this if to reduce nesting...

Reply:
Changed.